src/xdp/program.c

```diff
@@ -691,6 +691,9 @@ XdpParseFrame(
             goto BufferTooSmall;
         }
         Cache->Ip4Hdr = (IPV4_HEADER *)&Va[Offset];
+        if (Cache->Ip4Hdr->HeaderLength != sizeof(*Cache->Ip4Hdr)) {
```

Comment:
We also need a similar check in `XdpParseFragmentedIp4` for consistency. I think it would be something like:

```
    Cache->Ip4Valid =
        XdpGetContiguousHeader(
            Frame, Buffer, BufferDataOffset, FragmentIndex, FragmentsRemaining, FragmentRing,
            VirtualAddressExtension, &Storage->Ip4Hdr, sizeof(Storage->Ip4Hdr), &Cache->Ip4Hdr) &&
        (Cache->Ip4Hdr->HeaderLength >> 2) == sizeof(*Cache->Ip4Hdr);
```

Reply:
Just re-pushed with that fix too (but left shift `<< 2` instead of right shift 😉)